stdlib/fun.mli

```diff
@@ -61,3 +63,81 @@ exception Finally_raised of exn
     an unexpected exception or a programming error. As a general rule,
     one should not catch a [Finally_raised] exception except as part of
     a catch-all handler. *)
+
+(** {1:examples Examples}
+
+    {2 Combinators}
+
+    {{!combinators}Combinators} provide a lightweight and sometimes more
+    readable way to create anonymous functions, best used as short-lived
+    arguments rather than standalone definitions. The examples below will
+    demonstrate this mainly with the {!module:List} module.
+
+    {!val:id}
+    {[
+    # List.init 3 Fun.id;;
+    - : int list = [0; 1; 2]
+
+    # List.filter_map Fun.id [None; Some 2; Some 3; None; Some 5];;
+    - : int list = [2; 3; 5]
+
+    # let to_flat = Float.Array.map_from_array Fun.id
+    val to_flat : float array -> Float.Array.t
+    ]}
+    Dispatching functions of type [foo -> foo] conditionally is another place
+    where [id] may be useful
+    {[
+    if Sys.win32 then String.map (function '\\' -> '/' | c -> c) else Fun.id
+    ]}
+
+    {!val:const}
+    {[
+    # List.init 3 (Fun.const 0);;
+    - : int list = [0; 0; 0]
+
+    # let last xs = List.fold_left (Fun.const Option.some) None xs;;
```

Comment:
I find this example obscure -- applying Fun.const to a one-element to a two-element function is weird. I would write it `fun _acc x -> Some x` or something. The `List.init` example above makes more sense to me.

Reply:
Quick ack & response until I can re-examine the code with fresh eyes: the idea was to progress into more "fun" examples to give the user a feel of the capabilities of those functions, with the most straightforward example being the first. I could just keep one straightforward example for each function, two at most, and avoid this delights style entirely.

Reply:
I would propose the new `String.spellcheck` function as a natural use case for `Fun.const`.

Reply:
Addressed by removing examples that are too obscure in terms of combinator usage, and indicating new "advanced" examples which try to be straightforward in terms of combinator usage, but more elaborate in utility. Also added caution paragraphs.